### PR TITLE
Feat: Create CCL API getDCCAdmissionCheckScenarios (EXPOSUREAPP-11990)

### PR DESCRIPTION
**Pull request by T-Bendrath**
This PR adds a new CCL API used to retrieve available admission check scenarios. Exemplary data for the scenarios DE, BW and HH are included in the PR.

Jira Backend Reference: EXPOSUREBACK-2090

### CODE REVIEW

**Review by CV113 — APPROVED** [review]
UA approved

**Comment by mlenkeit** [comment]
Changing base branch to avoid merging into `main` for now.

File: lib/ccl/functions/getDccAdmissionCheckScenarios.js [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+                      'scenarioSelectionItem',
+                      {
+                        init: [
+                          'object',
+                          'identifier', {
+                            init: [
+                              'literal',
+                              { var: 'it.identifier' }
+                            ]
+                          },
```
Comment (mlenkeit) [inline lib/ccl/functions/getDccAdmissionCheckScenarios.js:89]: no need for init here, just `'identifier', { var: 'it.identifier' }` should do

File: lib/ccl/functions/getDccAdmissionCheckScenarios.js [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+                              '__i18n.getTextDescriptor',
+                              {
+                                key: {
+                                  concatenate: [
+                                    'ADMISSION_SCENARIO_SELECTION_',
+                                    {
+                                      toUpperCase: [
+                                        { var: 'it.identifier' }
+                                      ]
+                                    },
```
Comment (mlenkeit) [inline lib/ccl/functions/getDccAdmissionCheckScenarios.js:135]: I would suggest to drop `toUpperCase`, the identifiers are already upper case

File: resources/i18n/de/strings.xml [hunk 3]
```diff
@@ -1,6 +1,26 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <resources xmlns:tools="http://schemas.android.com/tools" xmlns:xliff="urn:oasis:names:tc:xliff:document:1.2" tools:ignore="MissingTranslation">
 
+  <!-- XHED: Admission Scenario - Card Title -->
+  <string name="ADMISSION_SCENARIO_LABEL_TEXT">"Status für folgendes Bundesland"</string>
+  <!-- XTXT: Admission Scenario - Card Title -->
```
Comment (mlenkeit) [inline resources/i18n/de/strings.xml:6]:
```suggestion
  <!-- XTXT: Admission Scenario - Card Subtitle Text -->
```

**Review by mlenkeit — CHANGES_REQUESTED** [review]

File: resources/i18n/de/strings.xml (hunk 3 above)
Comment (T-Bendrath) [reply resources/i18n/de/strings.xml]: Resolved in https://github.com/corona-warn-app/cwa-app-ccl/commit/e6618f359606a912fdaead0db0dc1ea5c994f513

File: lib/ccl/ccl-main.js [hunk 4]
```diff
@@ -64,6 +68,14 @@ export default {
...
         }
       },
+      getDccAdmissionCheckScenarios: {
+        input: {
+          validate: instance => validateSchema(instance, 'https://ccl.coronawarn.app/functions/getDccAdmissionCheckScenarios#/$defs/input')
+        },
+        output: {
+          validate: instance => validateSchema(instance, 'https://ccl.coronawarn.app/functions/getDccAdmissionCheckScenarios#/$defs/output')
+        }
+      },
```
[the remark this answers is not in the document]
Comment (T-Bendrath) [reply lib/ccl/ccl-main.js]: Proposed changes are implemented in https://github.com/corona-warn-app/cwa-app-ccl/pull/36/commits/45307f785774aa63b5d8797f8d217ef38b1edca6

File: lib/ccl/ccl-main.js [hunk 5]
```diff
@@ -46,6 +47,9 @@ export default {
   api: {
     getDccWalletInfo: input => {
       return evaluateFunction('getDccWalletInfo', input)
+    },
+    getDccAdmissionCheckScenarios: input => {
+      return evaluateFunction('getDccAdmissionCheckScenarios', input)
     }
```
[the remark this answers is not in the document]
Comment (T-Bendrath) [reply lib/ccl/ccl-main.js]: Proposed changes are implemented in https://github.com/corona-warn-app/cwa-app-ccl/pull/36/commits/45307f785774aa63b5d8797f8d217ef38b1edca6

File: lib/ccl/ccl-jfn.js [hunk 6]
```diff
@@ -30,6 +31,7 @@ const functions = [
   __sortVaccinationCertificatesByDate,
   __toCertificateReference,
   getDccWalletInfo,
+  getDccAdmissionCheckScenarios,
```
[the remark this answers is not in the document]
Comment (T-Bendrath) [reply lib/ccl/ccl-jfn.js]: Proposed changes are implemented in https://github.com/corona-warn-app/cwa-app-ccl/pull/36/commits/45307f785774aa63b5d8797f8d217ef38b1edca6

File: lib/ccl/ccl-jfn.js [hunk 7]
```diff
@@ -13,6 +13,7 @@ import __sortTestCertificatesByDate from './functions/__sortTestCertificatesByDa
 import __sortVaccinationCertificatesByDate from './functions/__sortVaccinationCertificatesByDate.js'
 import __toCertificateReference from './functions/__toCertificateReference.js'
 import getDccWalletInfo from './functions/getDccWalletInfo.js'
+import getDccAdmissionCheckScenarios from './functions/getDccAdmissionCheckScenarios.js'
```
[the remark this answers is not in the document]
Comment (T-Bendrath) [reply lib/ccl/ccl-jfn.js]: Proposed changes are implemented in https://github.com/corona-warn-app/cwa-app-ccl/pull/36/commits/45307f785774aa63b5d8797f8d217ef38b1edca6

File: lib/ccl/functions/getDccAdmissionCheckScenarios.js [hunk 8]
```diff
@@ -0,0 +1,176 @@
...
+          {
+            call: [
+              '__i18n.getTextDescriptor',
+              {
+                key: 'ADMISSION_SCENARIO_LABEL_TEXT'
+              }
+            ]
+          }
+        ]
+      },
```
[the remark this answers is not in the document]
Comment (T-Bendrath) [reply lib/ccl/functions/getDccAdmissionCheckScenarios.js]: Proposed changes are implemented in https://github.com/corona-warn-app/cwa-app-ccl/pull/36/commits/b1c5d7032a735ead55265e051836a16015d33d0a

File: lib/ccl/functions/getDccAdmissionCheckScenarios.js (hunk 1 above)
Comment (T-Bendrath) [reply lib/ccl/functions/getDccAdmissionCheckScenarios.js]: Proposed changes are implemented in https://github.com/corona-warn-app/cwa-app-ccl/pull/36/commits/b1c5d7032a735ead55265e051836a16015d33d0a

**Review by mlenkeit — APPROVED** [review]
LGTM

**Pull request merged**